ci/dart.yml

```
@@ -10,18 +15,13 @@ jobs:
   build:
     runs-on: ubuntu-latest
 
-    # Note that this workflow uses the latest stable version of the Dart SDK.
-    # Docker images for other release channels - like dev and beta - are also
-    # available. See https://hub.docker.com/r/google/dart/ for the available
-    # images.
-    container:
-      image:  google/dart:latest
-
     steps:
       - uses: actions/checkout@v2
 
-      - name: Print Dart SDK version
-        run: dart --version
+      # Note: This workflow uses the latest stable version of the Dart SDK.
+      # You can specify other versions if desired, see documentation here:
+      # https://github.com/dart-lang/setup-dart/blob/main/README.md
+      - uses: dart-lang/setup-dart@v1
```

Looks good, but I think we'll have to use the 40 char SHA of `setup-dart` for the moment.

---

Using SHAs seems like a poor user experience compared to versions. Is there no other way around this?

---

@andymckay, is SHAs a hard requirement?

---

Outside of some code scanning workflows and actions in the https://github.com/actions Organization, this is a requirement. We're working on updating our Contributing guidelines to clarify this.

I agree this is not a great user experience, I'd recommend adding a comment above the `uses` for users to easily switch to a versioned tag. We do something similar in https://github.com/actions/starter-workflows/blob/main/ci/ruby.yml

```suggestion
      # - uses: dart-lang/setup-dart@v1
      - uses: dart-lang/setup-dart@9a04e6d73cca37bd455e0608d7e5092f881fd603
```

---

OK, thanks. Took the commit suggestion.